pkg/operator/sync_openshiftcontrollermanager_v311_00.go

```diff
@@ -295,5 +299,94 @@ func manageOpenShiftControllerManagerDeployment_v311_00_to_latest(client appscli
 	}
 	required.Annotations[util.VersionAnnotation] = os.Getenv("RELEASE_VERSION")
 
+	proxyCfg, err := proxyLister.Get("cluster")
+	if err != nil {
+		recorder.Eventf("ProxyConfigGetFailed", "Error retrieving global proxy config: %s", err.Error())
+		if !apierrors.IsNotFound(err) {
+			// return daemonset since it is still referenced by caller even with errors
+			return required, false, err
+		}
+	}
+	if err == nil {
+		noProxySet := len(proxyCfg.Status.NoProxy) > 0
+		httpProxySet := len(proxyCfg.Status.HTTPProxy) > 0
+		httpsProxySet := len(proxyCfg.Status.HTTPSProxy) > 0
+
+		for i, c := range required.Spec.Template.Spec.Containers {
+			newEnvs := []corev1.EnvVar{}
+			hasNoProxy := false
+			hasHTTPProxy := false
+			hasHTTPSProxy := false
+			for _, env := range c.Env {
+				name := strings.TrimSpace(env.Name)
+				if name != "NO_PROXY" && name != "HTTPS_PROXY" && name != "HTTP_PROXY" {
+					// add any non proxy env to the new list
+					newEnvs = append(newEnvs, env)
+					continue
+				}
+				if name == "NO_PROXY" {
+					if !noProxySet {
+						// if no longer set in config, do not add to new list (i.e. delete), force rollout
+						forceRollout = true
+						continue
+					}
+					// note already exists, re-add to new list, with up to date value
+					hasNoProxy = true
```

Comment:
This asks to be a switch, not a set of ifs.

```go
switch name {
case "NO_PROXY":
    ...
case "HTTPS_PROXY":
    ...
case "HTTP_PROXY":
    ...
default:
   newEnvs = append(newEnvs, env)
}

Reply:
update pushed in separate commit thanks @soltysh